lib/saluki-env/src/autodiscovery/mod.rs

```diff
@@ -96,6 +99,49 @@ pub struct Config {
     pub logs_excluded: bool,
 }
 
+impl Config {
+    /// Get the ID for an instance
+    pub fn id(&self, digest: &u64, instance: &HashMap<MetaString, serde_yaml::Value>) -> String {
```

Comment:
Separately: what's the _point_ of this function?

My naive thought is that the config instance should be wrapped up in some sort of helper type -- let's call it `ConfigInstance` -- where it holds both the raw instance config _and_ the ID for that instance.

Doing that would allow you to handle all of the error checking, which is being skipped here, as part of initially creating the `Config`, instead of having to do it all on-demand.

Reply:
That could work. 

The idea of having digest and id functions separated is to avoid computing the digest multiple times for checks with multiple instances. 

> Doing that would allow you to handle all of the error checking, which is being skipped here, as part of initially creating the Config, instead of having to do it all on-demand.

I guess we could do it, but doing it on demand means that if we receive non-check events from the autodiscovery provider, we can skip them without doing any extra computation. BTW, that event skipping is not implemented yet

Reply:
If we're receiving non-check events, though... then they wouldn't have anything in `instances`, no?

That's mostly what I'm getting at: if we've decoded an event far enough that we know it has check config instances in `instances`, then we could go one small step further during the creation of `Config` and wrap that up in some hypothetical type like `ConfigInstance`, and we'd only be computing the digest/ID when creating `ConfigInstance`, because we know we always need to query those values at least once.

Reply:
I've decoded the config further and split between checks config and generic configuration events. I added two specific legs to the `AutodiscoveryEvent` enum to differentiate between generic and check events as well. 

Here is the code https://github.com/DataDog/saluki/pull/662/commits/16d66ab224e510dd5236cb852399ee300f7f1c03

Please let me know if that's what you think.